### PR TITLE
Documentation simply 4150

### PR DESCRIPTION
**Pull request by thinkwebengineer**
## Description

Documentation of admin API end points for ticket 4150

## Motivation and Context

To improve utilization with proper API documentation.

## Checklist:

- [x] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: api/documentation/controller.py [hunk 1]
```diff
@@ -405,6 +405,95 @@ def addSchemes(self):
...
+            {
+                'admin_auth_services': {'$ref': '#/components/schemas/AuthServicesDictSchema'},
+                'protocols': {'$ref': '#/components/schemas/ProtocolDictSchema'}
+            }
+        )
+
+        self.addComponent(
+            'schema', 'AdminAuthPost', 'object',
+            {
+                'protocol': {'type': 'string'},
```
Comment (mwbenowitz) [inline api/documentation/controller.py]: I think this should be some kind of JSON object. We should capture that even if we don't fully capture everything that should be submitted here. The general shape of what the protocol should look like would be really useful!

File: api/admin/routes.py [hunk 2]
```diff
@@ -803,6 +936,58 @@ def patron_auth_self_tests(identifier):
 @requires_admin
 @requires_csrf_token
 def lookup_patron():
+    """Look up personal information about a patron via the ILS
```
Comment (mwbenowitz) [inline api/admin/routes.py:939]: The `200` response code says "A confirmation that the password was updated" which doesn't match this description. Could you clarify specifically what this does? The fact that it's a POST makes me feel like it is actually probably doing something like resetting a password.

File: api/admin/routes.py [hunk 3]
```diff
@@ -812,6 +997,60 @@ def lookup_patron():
...
+            type: string
+      requestBody:
+        required: true
+        content:
+          multipart/form-data:
+            schema: String
+            example: Identifier for patron
+      responses:
+        200:
+          description: A text confirmation that the password was updated
```
Comment (mwbenowitz) [inline api/admin/routes.py]: Should be a text confirmation that the Adobe ID was successfully reset!

File: api/admin/routes.py [hunk 4]
```diff
@@ -965,6 +1204,94 @@ def sitewide_setting(key):
...
+        Create or update logging services and associated protocols.
+      security:
+        - BasicAuth: []
+      parameters:
+        - X-CSRF-Token
+      requestBody:
+        required: true
+        content:
+          multipart/form-data:
+            schema: AdminAuthPost
```
Comment (mwbenowitz) [inline api/admin/routes.py:1257]: Is this the same as the `Protocol` object used for the `admin_auth_services` requests? I'm not really familiar with either of these endpoints!

**Review by mwbenowitz — APPROVED** [review]
A few questions but otherwise looks great!

File: api/documentation/controller.py (hunk 1 above)
Comment (thinkwebengineer) [reply api/documentation/controller.py]: This is for the POST request on admin_auth_services and is grabbing a protocol name from a form to then search for the protocols objects.

File: api/admin/routes.py (hunk 4 above)
Comment (thinkwebengineer) [reply api/admin/routes.py:1257]: These are asking for the same information from the requests in 2 different controller areas. the protocol that is asked for in the form is a string that is used to search protocols by name.

File: api/documentation/controller.py (hunk 1 above)
Comment (thinkwebengineer) [reply api/documentation/controller.py]: Added description in the schema for clarity.

**Pull request merged**